src/python/pants/backend/python/util_rules/pex_from_targets.py

```diff
@@ -312,6 +312,28 @@ async def pex_from_targets(request: PexFromTargetsRequest, python_setup: PythonS
             "`[python-setup].requirement_constraints` must also be set."
         )
 
+    if python_setup.lockfile:
```

Comment:
Is this supposed to be an `elif` on the if statement above? It looks like the effect of this currently is that we'll set the local variable `repository_pex` multiple times...

Reply:
No need, we've already validated that lockfile is not set at the same time as constraints. But I can adjust that to make it even more clear in a followup.